compiler/rustc_hir_analysis/src/astconv/errors.rs

```diff
@@ -597,7 +597,28 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                 }
             }
         }
-        if !suggestions.is_empty() {
+        suggestions.sort_by_key(|&(span, _)| span);
```

Comment:
Are you certain that the suggestion spans will end up sorted in such a way that two non-adjacent keys overlap? If not, then since this is the error path, I feel like it may just be easier to do the O(N^2) full pairwise comparison here.

Reply:
`SpanData` manually implements `Ord` on `(lo, hi, ctxt)`. That means that by sorting any two adjacent spans that don't overlap imply that the spans further apart from each other also don't overlap.

Reply:
What about spans that overlap only at their prefix? Like spans that look like: `[(1, 5), (2, 3), (4, 5)]`, where the first and last elements overlap?

Reply:
Maybe it doesn't matter, since by construction that would imply any intermediate spans overlap. Idk.

Reply:
Your last comment is exactly right.